util/test/Test/Pos/Util/Golden.hs

```diff
@@ -38,6 +43,35 @@ goldenTestJSON x path = withFrozenCallStack $ do
             Left err -> failWith Nothing $ "could not decode: " <> show err
             Right x' -> x === x'
 
+-- | Only check that the datatype equals the decoding of the file
```

Comment:
Shouldn't we also be checking the encoding?
https://github.com/input-output-hk/cardano-sl/pull/3415

Reply:
The reason for these functions is to ensure backwards compatibility with existing serialized datatstructures, with default values for missing fields. We want to demonstrate that when the PR is merged, the tests for “deserialize this bytestring and check that the value is equal to a target value” still pass (once the given datatype is modified, in the source code, to have the default value in the newly added record field)